File: javascript/productSumArray.js

function productSumArray(array) {
  return productSumArrayHandle(array, 1)
}

function productSumArrayHandle(array, depth) {
  let sum = 0
  for (const number of array) {
    if (Array.isArray(number)) {
      sum += productSumArrayHandle(number, depth + 1)
    } else {
      sum += number
    }
  }
  return depth * sum
}

module.exports = productSumArray

File: python/productSumArray.py

def productSumArray( array ):
  return productSumArrayHandle( array, 1 )

def productSumArrayHandle( array, depth ):
  sum = 0
  for number in array:
    if isinstance(number, list):
      sum += productSumArrayHandle( number, depth + 1 )
    else:
      sum += number

  return depth * sum

print(productSumArray([1,2,[3,4]]))
print(productSumArray([1,2,[3,4],[1,2,[1,2]]]))
print(productSumArray([1,2,[3,4],[1,2,[1,2]],[]]))
